pkg/prometheus/instance/instance.go

```diff
@@ -339,15 +287,32 @@ func (i *Instance) Run(ctx context.Context) error {
 		)
 	}
 
-	err = g.Run()
+	err = rg.Run()
 	if err != nil {
 		level.Error(i.logger).Log("msg", "agent instance stopped with error", "err", err)
 	}
 	return err
 }
 
-func (i *Instance) newDiscoveryManager(ctx context.Context) (*discovery.Manager, error) {
-	manager := discovery.NewManager(ctx, log.With(i.logger, "component", "discovery manager scrape"), discovery.Name("scrape"))
+type discoveryService struct {
+	RunFunc    func() error
+	StopFunc   func(err error)
+	SyncChFunc func() GroupChannel
+}
+
+func (s *discoveryService) Run() error           { return s.RunFunc() }
+func (s *discoveryService) Stop(err error)       { s.StopFunc(err) }
+func (s *discoveryService) SyncCh() GroupChannel { return s.SyncChFunc() }
+
+// newDiscoveryManager returns an implementation of a runnable service
+// that outputs discovered targets to a channel. The implementation
+// uses the Prometheus Discovery Manager. Targets will be filtered
+// if the instance is configured to perform host filtering.
+func (i *Instance) newDiscoveryManager(ctx context.Context) (*discoveryService, error) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	logger := log.With(i.logger, "component", "discover manager scrape")
```

Comment:
final nitpick: component value is a little weird here. `service discovery manager` + trimming the logs in the `rg,Add` to be less redundant.